boofuzz/pedrpc.py

```diff
@@ -42,9 +42,9 @@ def __connect(self):
         self.__disconnect()
 
         # connect to the server, timeout on failure.
+        self.__server_sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        self.__server_sock.settimeout(3.0)
         try:
-            self.__server_sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-            self.__server_sock.settimeout(3.0)
             self.__server_sock.connect((self.__host, self.__port))
```

Comment:
@SR4ven Did pytype flag this one because more code was in the `try` than necessary? If so that's pretty neat.

Reply:
No it's not that smart sadly.
The error here is:
```python
pedrpc.py, line 62, in __connect: No attribute 'setsockopt' on None [attribute-error]
  In Optional[socket.socket]
```
Pytype assumes that the assignment of `self.__server_sock` fails, which could be caught by the try/except. That again would let the program continue to `self.__server_sock.settimeout(None)` which would then fail because the variable is not an instance of socket.

Reply:
Ah, thanks -- still a nice catch.